crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -663,7 +696,17 @@ fn highlight_name(db: &RootDatabase, def: Definition) -> Highlight {
                 if func.is_unsafe(db) {
                     h |= HighlightModifier::Unsafe;
                 }
-                return h;
+                return if func.has_self_param(db) {
```

Comment:
Can I just mark `hir::ModuleDef::Function(_) => unreachable!()`? I believe this entire block is covered by the call to `highlight_func_by_name_ref` in `highlight_element`, but I wasn't sure if this would be used somewhere else. This does not cover the `Consuming` cases.